src/main/java/com/google/collegeplanner/data/Course.java

```diff
@@ -101,14 +96,33 @@ public Course(String courseId, String name, String semester, int credits, String
   }
 
   public Course(JSONObject json) {
-    this((String) json.get("course_id"), (String) json.get("name"), (String) json.get("semester"),
-        (int) json.get("credits"), (String) json.get("dept_id"), (String) json.get("description"),
-        (String[]) ((JSONArray) json.get("grading_method")).toArray(),
-        (String) ((JSONObject) json.get("relationships")).get("coreqs"),
-        (String) ((JSONObject) json.get("relationships")).get("prereqs"),
-        (String) ((JSONObject) json.get("relationships")).get("restrictions"),
-        (String) ((JSONObject) json.get("relationships")).get("additional_info"),
-        (String) ((JSONObject) json.get("relationships")).get("credit_granted_for"));
+    try {
+      this.credits = Integer.parseInt((String) json.get("credits"));
+    } catch (Exception e) {
+      this.credits = 0;
+    }
+
+    this.courseId = (String) json.get("course_id");
```

Comment:
do any of these fields need to be validated? are empty fields values for any of these fields ok? or should you throw your own error if you encounter an incomplete object?

Reply:
description, prereqs, coreqs, etc are oftentimes null. I don't think those matter that much, but a null course_id for example would be bad. So we'd maybe want to skip it? Not sure